strawberry/field.py

```diff
@@ -234,17 +287,22 @@ def copy_with(
             default_factory=self.default_factory,  # type: ignore[misc]
             deprecation_reason=self.deprecation_reason,
         )
+        field.type = new_type
+        return field
 
     def get_result(
-        self, source: Any, info: Info, args: List[Any], kwargs: Dict[str, Any]
+        self, source: Any, info: Info, arguments: Dict[str, Any]
     ) -> Union[Awaitable[Any], Any]:
```

Comment:
Should we give this function a better name, esp. if users are expected to override it?

Reply:
Any suggestions? I don't think `get_result` is a bad name.